sixtyfps_runtime/corelib/layout.rs

```diff
@@ -82,30 +72,24 @@ impl LayoutInfo {
         };
 
         Self {
-            min_width: self.min_width.max(other.min_width),
-            max_width: self.max_width.min(other.max_width),
-            min_height: self.min_height.max(other.min_height),
-            max_height: self.max_height.min(other.max_height),
-            min_width_percent: self.min_width_percent.max(other.min_width_percent),
-            max_width_percent: self.max_width_percent.min(other.max_width_percent),
-            min_height_percent: self.min_height_percent.max(other.min_height_percent),
-            max_height_percent: self.max_height_percent.min(other.max_height_percent),
-            preferred_width: merge_preferred_size(
-                self.horizontal_stretch,
-                self.preferred_width,
-                other.horizontal_stretch,
-                other.preferred_width,
+            min: self.min.max(other.min),
+            max: self.max.min(other.max),
+            min_percent: self.min_percent.max(other.min_percent),
+            max_percent: self.max_percent.min(other.max_percent),
+            preferred: merge_preferred_size(
+                self.stretch,
+                self.preferred,
+                other.stretch,
+                other.preferred,
             ),
-            preferred_height: merge_preferred_size(
-                self.vertical_stretch,
-                self.preferred_height,
-                other.vertical_stretch,
-                other.preferred_height,
-            ),
-            horizontal_stretch: self.horizontal_stretch.min(other.horizontal_stretch),
-            vertical_stretch: self.vertical_stretch.min(other.vertical_stretch),
+            stretch: self.stretch.min(other.stretch),
         }
     }
+
+    /// Helper function to return a preferred size which is within the min/max constraints
+    pub fn preferred(&self) -> f32 {
```

Comment:
Hn, so `info.preferred` and `info.preferred()` look really alike but they evaluate to something rather different. Maybe the function should be called `preferred_constrained()` or `constrained_preferred_size()` or `bounded_preferred_size()`.

Reply:
I'll rename it